src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -496,7 +496,13 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                     }
                 }
 
-
+                var arch = await MICommandFactory.GetTargetArchitecture();
+                if (arch == TargetArchitecture.Unknown)
```

Comment:
Ug. How about this then - ask the debugger for its target architecture, if we understand it -- good. If not, check if it is in the launch options. If it is -- use that. If it is missing from both places, send a warning to the output window and guess x64.